x-pack/plugins/uptime/server/lib/synthetics_service/hydrate_saved_object.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UptimeESClient } from '../lib';
+import { UptimeServerSetup } from '../adapters';
+import { SyntheticsMonitorSavedObject } from '../../../common/types';
+import { MonitorFields, Ping } from '../../../common/runtime_types';
+
+export const hydrateSavedObjects = async ({
+  monitors,
+  server,
+}: {
+  monitors: SyntheticsMonitorSavedObject[];
+  server: UptimeServerSetup;
+}) => {
+  const missingUrlInfoIds: string[] = [];
+
+  monitors
+    .filter((monitor) => monitor.attributes.type === 'browser')
+    .forEach(({ attributes, id }) => {
+      const monitor = attributes as MonitorFields;
+      if (!monitor || !monitor.urls || monitor.urls.length === 0) {
```

Comment:
I think we need to also add in `|| monitor.type === DataStream.BROWSER`, in order to ensure an update path as well. Or we need to track revisions some how and make sure that this gets called after a monitor is edited. Perhaps set `urls`, back to `undefined` when a monitor is edited.

Reply:
@dominiqueclarke  yeah i think setting it to `undefined` shoud do the trick 

`monitor.type === DataStream.BROWSER` that filter is already added see the line 23